ETL/Python ETL/MatchHistoryETL.py

```diff
@@ -0,0 +1,210 @@
+    print (f'Can not connect to the postgress database "{constants.DB_NAME}". Make sure database server is running')
+    print (e)
+else:
+    print (f'Connection to database "{constants.DB_NAME}" stablished. Listening at port {constants.DB_PORT}')
+
+
+# Find out current season matchweek
+cursor = connection.cursor()
+next_matchweek = calculate_matchweek(cursor)
+
+
+# Add MatchWeek column
+
+website_frame.insert(2, "MatchWeek", next_matchweek)
+
+# Add Points columns
+
+conditions = [
+     website_frame["FullTimeResult"] == 'H',
+     website_frame["FullTimeResult"] == 'D',
+     website_frame["FullTimeResult"] == 'A'
+]
+
+home_points = [ 3, 1, 0]
+away_points = [ 0, 1, 3]
+
+website_frame["HomeTeamPoints"] = np.select(conditions, home_points)
+website_frame["AwayTeamPoints"] = np.select(conditions, away_points)
+
+
+def generate_upsert_stmn(table_name, conflict_column, dataframe):
+    """
+        Generates a sql statement for Postgresql SQL dialect equivalent to UPSERT operation.
+        This is achieved by writing standard INSERT clause along with ON CONFLICT
+        Resulting statement adopts posture of skipping insertions on rows that conflict with
+        conflict_column field
+
+        dataframe - source dataframe which contains data to be inserted
+        conflict_column - collumn considered to avoid any duplications
+        table_name - Schema Table name to generate insert values
+    """
+    def entitle_value(value):
+        if type(value) is str:
+            return "'" + value + "'"
+        # Return stringified version of anything else (Numbers, Dates, Floats)
+        return str(value)
+
+    def generate_insert_stmn(table_name, dataframe):
+        entitled_headers = [ '"' + column_name + '"' for column_name in  dataframe.columns ]
+        return "INSERT INTO " + table_name + " (" + ", ".join(entitled_headers) + ") "
+    
+    def generate_values_stmn(row):
+        values = row[1].values
+        values_stmt = " ("
+        value_idx = 1
+        for value in values:
+            values_stmt += entitle_value(value)
+            value_idx += 1
+            # Skip last comma in statement
+            if value_idx <= len(values):
+                values_stmt +=  ", "
+    
+        values_stmt +=")"
+        return values_stmt
+
+        
+    sql_statement = ""
+    sql_statement += generate_insert_stmn(table_name, dataframe)
+    sql_statement += "VALUES "
+    value_idx = 1
+    for row in dataframe.iterrows():
+        sql_statement += generate_values_stmn(row)
+        value_idx += 1
+        # Skip comma on last row
+        if value_idx <= dataframe.shape[0]:
+            sql_statement += ", "
+        
+    sql_statement +=" ON CONFLICT (\"" + conflict_column + "\")  DO NOTHING"
+    return sql_statement
+
+
+insert_statement = generate_upsert_stmn("public.match_history", "MatchID", website_frame)
+print(insert_statement)
+
+
+# !! Getting error here
+cursor.execute(insert_statement)
```

Comment:
Getting error message due to quote conflict when inserting values to database like "Nott'm Forest"

Reply:
Nevermind, I solved this issue by using psycopg2.sql package, it offers very handy string composition tools to build up your SQL query from scratch, taking care of escaping and security concerns.